identifyyeastcells.py

```diff
@@ -0,0 +1,1146 @@
+                 self.autoadaptation_steps,
+                 self.use_ground_truth_to_set_params,
+                 self.show_autoadapted_params]
+
+        if self.show_autoadapted_params:
+            list.append(self.autoadapted_params)
+
+        list.append(self.advanced_parameters)
+
+        #
+        # Show the user the background only if self.provide_background is checked
+        #
+        if self.advanced_parameters:
+            list.append(self.segmentation_precision)
+
+            list.append(self.specify_precision_details)
+            if self.specify_precision_details:
+                list.append(self.iterations)
+                list.append(self.seeds_border)
+                list.append(self.seeds_content)
+                list.append(self.seeds_centroid)
+                list.append(self.contour_points)
+                list.append(self.contour_precision)
+                list.append(self.weights_number)
+
+            list.append(self.maximal_cell_overlap)
+            list.append(self.advanced_cell_filtering)
+            if self.advanced_cell_filtering:
+                list.append(self.min_cell_area)
+                list.append(self.max_cell_area)
+            # Show the user the background only if self.provide_background is checked
+            list.append(self.background_elimination_strategy)  #
+
+            if self.background_elimination_strategy == BKG_FILE:
+                list.append(self.background_image_name)
+
+            list.append(self.ignore_mask_image_name)
+
+        list.append(self.should_save_outlines)
+        #
+        # Show the user the scale only if self.should_save_outlines is checked
+        #
+        if self.should_save_outlines:
+            list.append(self.save_outlines)
+
+        return list
+
+
+    def on_setting_changed(self, setting, pipeline):
+        '''If precision is changed then update all the related settings'''
+        if setting == self.segmentation_precision or \
+                                setting == self.specify_precision_details and not self.specify_precision_details.value:
+            self.set_ui_from_precision(self.segmentation_precision.value)
+
+
+    def is_interactive(self):
+        return False
+
+    def prepare_group(self, workspace, grouping, image_numbers):
+        '''Erase module information at the start of a run'''
+        d = self.get_dictionary(workspace.image_set_list)
+        d.clear()
+        return True
+
+    def get_ws_dictionary(self, workspace):
+        return self.get_dictionary(workspace.image_set_list)
+
+    def __get(self, field, workspace, default):
+        if self.get_ws_dictionary(workspace).has_key(field):
+            return self.get_ws_dictionary(workspace)[field]
+        return default
+
+    def __set(self, field, workspace, value):
+        self.get_ws_dictionary(workspace)[field] = value
+ 
+    def upgrade_settings(self, setting_values, variable_revision_number, module_name, from_matlab):
```

Comment:
(This is an optional recommendation.)

Since this module has not existed in CellProfiler-plugins before, you can remove `upgrade_settings` and start fresh. You may have to modify your existing pipelines, but it's less code to maintain going forward.